### ISSUE TITLE
2 units on output layer for a dichotomy classifier

### ISSUE THREAD

**Issue by adailsonfilho** [aigamedev/scikit-neuralnetwork#210]
Hi guys,

I'm got myself in a situation that I didn't found an explanation in the docs. I'm set up a MLP classifier form a database with 2 classes {0,1} and around 10.000 samples, but I was forced to use 2 units in the output layer, that obviously had no need, since i'm trying to solve a dichotomy problem, 2 classes, 1 unit activeted or not suit well. I've tracked down the error and noticed that it was intentional, ok, but I don't undestand why explicity these 2 units in the outputlayer in this case. Even the output is caming out like just one unit. Thanks guys, keep up the good work.

\o

**Comment by alexjc** [comment]
Having two units makes the code a bit simpler, there are no special cases for binary vs. N-classes.

**Comment by adailsonfilho** [comment]
Yeah,

I thought that was the case. Well, just my point of view: as a UX designer, and for matters of coherence with the neural network math models, I would recomend make it implicitly, because if it was only a manuever for simplify the code, the "user" is not obliged to know that. It makes the use flow better ;) ...Thanks again for make this available

\o

**Comment by alexjc** [comment]
As a user, you're given the classes as labels and the probabilities that you can just index. I think that's by far simplest...

**Comment by adailsonfilho** [comment]
I agree with you about what you just said. But I am talking about a more specific case, when you set up the output layer with 1 unit, explicity, that shouldn't result in an error for a 2 classes problem. Anyway, just my opinion. Pretty good job with the project! =)

\o

**Comment by alexjc** [comment]
Oh, I see! ;-) I never specify the units... maybe it should throw an exception if it mismatches?

**Comment by adailsonfilho** [comment]
Actually, I think that would be better if, in this case in particular, implicitly change the units to '2' without throwing any error or warning the user, and in the verbose log show as 1 unit as well, because this is what we really need to solve dichotomys. I think that should be explained in the code, with comments, so, just who want to know code details see it, as the code manuever that it is. Got it?
